data/items.lua

```diff
@@ -182,6 +186,7 @@ function items:AttachItemInfo(data)
   local effectiveIlvl, isPreview, baseIlvl = GetDetailedItemLevelInfo(itemID)
   data.containerInfo = C_Container.GetContainerItemInfo(bagid, slotid)
   data.questInfo = C_Container.GetContainerItemQuestInfo(bagid, slotid)
+  data.tooltipData = C_TooltipInfo.GetBagItem(bagid, slotid)
```

Comment:
Same reason/logic as above on partitioning this as tooltip data. Let's consider exposing this as just raw item data -- no need to tell the developer it came from the tooltip. I like indirection here, incase Blizzard adds an API down the line for this directly.

Reply:
I had originally isolated all of the tooltip code in the plugin/module, but it was added to the items.lua data structure per your discord request - I had thought you wanted it there to facilitate the data caching - but I'm unclear on item vs tooltip data-caching mechanics tbh.
One reason would be that data.tooltipData is available to plugin developers, so they could avoid further scanning/loading.

> Add binding info to the existing item properties 

The 'binding' data I added is the extracted, but un-categorized version. I was thinking it might be useful to have the raw data available to plugins. (Like bi-sectional armor filters for transmogs)
`    binding = itemTooltips:GetItemBinding(data.tooltipData) --[[@as Enum.TooltipDataItemBinding]],
`

GetItemBinding could be generalized into items.lua instead of itemtooltips.lua by moving the const.BINDING_MAP out of frames and into the tooltip module - more like equipmentsets are handled. I'll whip this up this weekend.

Note that the current code only supports retail and compatibility routines would be needed for Classic where C_Tooltip isn't implemented

Reply:
I don't think I was clear, sorry!

The reason I want to roll this into a data structure that isn't explicitly called out as tooltip is three fold:

1) I do actually believe that an API will come eventually. Blizzard in the last few years has been working to modernized a lot of the API's in general, and this is going to be an area of improvement for them.

2) I want to categorize data structures by their function, not by their source. For example, quest information comes from another API, but the item data structure doesn't mention that API, just that it's quest information.

3) Tooltip data needs further processing in all cases -- I want to provide this data already processed in the item data API, so that authors in the future don't have to reason about how to extract the data by making a call to itemtooltips.lua

I absolutely agree that itemtooltips.lua should remain -- this was a good idea and well thought out. The only change I'm asking for here is where the data lands in the `itemData` structure. Ideally, we would remove `data.tooltipData`, and only expose pre-calculated information for every item, i.e.

```lua
data.itemInfo.bound = SOME_ENUM
```

Does that make sense?